src/client_side_reply.cc

```diff
@@ -504,13 +509,17 @@ clientReplyContext::handleIMSReply(StoreIOBuffer result)
         debugs(88, 3, "origin replied with error " << status <<
                ", forwarding to client due to fail_on_validation_err");
         sendClientUpstreamResponse();
+        return;
     } else {
         // ignore and let client have old entry
         http->logType.update(LOG_TCP_REFRESH_FAIL_OLD);
         debugs(88, 3, "origin replied with error " <<
                status << ", sending old entry (" << oldStatus << ") to client");
         sendClientOldEntry();
+        return;
     }
+
+    // XXX: unreachable? or sending what to client?
```

Comment:
I see no reason to leave this important question unresolved in this PR. Unless I am missing something, we should remove all `else` clauses after `return` statements and then (naturally) remove this XXX. Long `else if` lists is rarely a good idea and it looks like it created real bugs in this particular case. I can do that if needed; just let me know.